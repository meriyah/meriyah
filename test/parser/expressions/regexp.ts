import { fail, pass } from '../../test-utils';

fail('RegExp', [String.raw`/\B*/u`]);

pass('RegExp', [{ code: String.raw`/\B*/u`, options: { validateRegex: false } }]);
