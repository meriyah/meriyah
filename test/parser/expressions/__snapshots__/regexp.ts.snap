// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RegExp > /\\B*/u 1`] = `
"SyntaxError [1:0-1:6]: Unterminated regular expression
> 1 | /\\B*/u
    | ^^^^^^ Unterminated regular expression"
`;

exports[`RegExp > /\\B*/u 2`] = `
{
  "body": [
    {
      "expression": {
        "regex": {
          "flags": "u",
          "pattern": "\\B*",
        },
        "type": "Literal",
        "value": null,
      },
      "type": "ExpressionStatement",
    },
  ],
  "sourceType": "script",
  "type": "Program",
}
`;
